lib/src/logic.dart

```diff
@@ -423,9 +423,15 @@ class Logic {
     }
   }
 
-  /// Accesses the [index]th bit of this signal.
-  Logic operator [](int index) {
-    return slice(index, index);
+  /// Accesses the [index]th bit of this signal, where index can be [int] or [Logic].
+  /// Throws Exception when index is not in range or index is neither [int] or [Logic].
+  Logic operator [](dynamic index) {
+    if (index is Logic) {
+      return IndexGate(this, index).selection;
+    } else if (index is int) {
+      return this.getRange(index, index + 1);
+    }
+    throw Exception('Expected [int] or [Logic]');
```

Comment:
I believe the bracket notation is used in doc comments, but not in Strings in things like Exceptions.  You could just put ticks around it instead:
```dart
    throw Exception('Expected `int` or `Logic`');
```